rhizome/kubernetes/lib/ubi_cni.rb

```diff
@@ -219,17 +219,23 @@ def error_exit(message)
     exit 1
   end
 
-  def find_random_available_ip(subnet)
-    allocated_ips = @ipam_store["allocated_ips"].values.flatten
-    max_retries = 100
+  def find_random_available_ip(subnet, reserved_ips: [])
+    allocated_ips = @ipam_store["allocated_ips"].values.flatten + reserved_ips
```

Comment:
Since `flatten` returns a new array, you can use `concat` to avoid an array allocation

```suggestion
    allocated_ips = @ipam_store["allocated_ips"].values.flatten.concat(reserved_ips)
```

Reply:
thanks